packages/ckeditor5-alignment/src/alignmentui.ts

```diff
@@ -127,28 +128,43 @@ export default class AlignmentUI extends Plugin {
 	private _addButton( option: SupportedOption ): void {
 		const editor = this.editor;
 
-		editor.ui.componentFactory.add( `alignment:${ option }`, locale => {
-			const command: AlignmentCommand = editor.commands.get( 'alignment' )!;
-			const buttonView = new ButtonView( locale );
-
-			buttonView.set( {
-				label: this.localizedOptionTitles[ option ],
-				icon: iconsMap.get( option ),
-				tooltip: true,
-				isToggleable: true
-			} );
+		editor.ui.componentFactory.add( `alignment:${ option }`, locale => this._createButton( locale, option ) );
+	}
 
-			// Bind button model to command.
-			buttonView.bind( 'isEnabled' ).to( command );
-			buttonView.bind( 'isOn' ).to( command, 'value', value => value === option );
+	/**
+	 * Helper method for creating the button view element.
+	 *
+	 * @param locale Editor locale.
+	 * @param option The name of the alignment option for which the button is added.
+	 * @param buttonAttrs Optional parameters passed to button view instance.
+	 */
+	private _createButton(
+		locale: Locale,
+		option: SupportedOption,
+		buttonAttrs: Partial<ExcludeMethodNames<ButtonView>> = {}
```

Comment:
I think there's a simpler way around this

```diff
diff --git a/packages/ckeditor5-alignment/src/alignmentui.ts b/packages/ckeditor5-alignment/src/alignmentui.ts
index 3d72bef237..d5a3a920ef 100644
--- a/packages/ckeditor5-alignment/src/alignmentui.ts
+++ b/packages/ckeditor5-alignment/src/alignmentui.ts
@@ -7,8 +7,8 @@
  * @module alignment/alignmentui
  */
 
-import { Plugin, icons, type ExcludeMethodNames } from 'ckeditor5/src/core.js';
-import { ButtonView, createDropdown, addToolbarToDropdown } from 'ckeditor5/src/ui.js';
+import { Plugin, icons } from 'ckeditor5/src/core.js';
+import { type Button, ButtonView, createDropdown, addToolbarToDropdown } from 'ckeditor5/src/ui.js';
 import type { Locale } from 'ckeditor5/src/utils.js';
 
 import { isSupported, normalizeAlignmentOptions } from './utils.js';
@@ -141,7 +141,7 @@ export default class AlignmentUI extends Plugin {
          private _createButton(
                   locale: Locale,
                   option: SupportedOption,
-                  buttonAttrs: Partial<ExcludeMethodNames<ButtonView>> = {}
+                  buttonAttrs: Partial<Button> = {}
          ): ButtonView {
                   const editor = this.editor;
                   const command: AlignmentCommand = editor.commands.get( 'alignment' )!;
diff --git a/packages/ckeditor5-ui/src/index.ts b/packages/ckeditor5-ui/src/index.ts
index a492d8d5d2..f73486cf67 100644
--- a/packages/ckeditor5-ui/src/index.ts
+++ b/packages/ckeditor5-ui/src/index.ts
@@ -19,6 +19,7 @@ export { default as BodyCollection } from './editorui/bodycollection.js';
 
 export { type ButtonExecuteEvent } from './button/button.js';
 export type { default as ButtonLabel } from './button/buttonlabel.js';
+export { default as Button } from './button/button.js';
 export { default as ButtonView } from './button/buttonview.js';
 export { default as ButtonLabelView } from './button/buttonlabelview.js';
 export { default as SwitchButtonView } from './button/switchbuttonview.js';
```

Reply:
Fixed